frontend/packages/topology/src/components/export-app/__tests__/ExportViewLogButton.spec.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+import { Button, Tooltip } from '@patternfly/react-core';
+import { mount } from 'enzyme';
+import { Provider } from 'react-redux';
+import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
+import { JobModel, PodModel } from '@console/internal/models';
+import store from '@console/internal/redux';
+import ExportViewLogButton from '../ExportViewLogButton';
+
+jest.mock('@console/internal/components/utils/k8s-watch-hook', () => ({
+  useK8sWatchResource: jest.fn(),
+}));
+
+const mockK8sWatchResource = useK8sWatchResource as jest.Mock;
+type Component = typeof ExportViewLogButton;
+type Props = React.ComponentProps<Component>;
+let ViewLogButtonProps: Props;
+
+describe('ExportViewLogButton', () => {
+  beforeEach(() => {
+    ViewLogButtonProps = { namespace: 'test' };
+    mockK8sWatchResource.mockImplementation((res) => {
+      if (!res) {
+        return [null, true, null];
+      }
+      switch (res?.kind) {
+        case PodModel.kind:
+          return [
+            {
+              kind: PodModel.kind,
+              metadata: { name: 'test', namespace: 'test' },
+              spec: { selector: { matchLabels: null } },
+            },
+            true,
+            null,
+          ];
+        case JobModel.kind:
+          return [
+            {
+              kind: JobModel.kind,
+              metadata: { name: 'test', namespace: 'test' },
+              spec: { selector: { matchLabels: null } },
+            },
+            true,
+            null,
+          ];
+        default: {
+          return [null, true, null];
+        }
+      }
+    });
+  });
+  it('should render a link and correct href path', () => {
+    // mount required for using provider
+    const logButtonWrapper = mount(
+      <Provider store={store}>
+        <ExportViewLogButton {...ViewLogButtonProps} />
+      </Provider>,
+    );
+    const logButton = logButtonWrapper.find('[data-test="export-view-log-btn"]').find(Button);
+    expect(logButton.prop('variant')).toBe('link');
+    expect(logButton.prop('component')).toBe('a');
+    expect(logButton.prop('href')).toBe('/k8s/ns/test/pods/test/logs');
+  });
+
+  it('should call onViewLog callback', () => {
+    const viewLogCallback = jest.fn();
+    ViewLogButtonProps.onViewLog = viewLogCallback;
+    // mount required for using provider
+    const logButtonWrapper = mount(
+      <Provider store={store}>
+        <ExportViewLogButton {...ViewLogButtonProps} />
+      </Provider>,
```

Comment:
Updating global is also more vulnerable to changes in other tests. Also if this is not an issue because you reset this here in `beforeEach`, but when it's easy to avoid this we should use simple local variables and props.

```suggestion
    const viewLogCallback = jest.fn();
    // mount required for using provider
    const logButtonWrapper = mount(
      <Provider store={store}>
        <ExportViewLogButton namespace="test" onViewLog={viewLogCallback} />
      </Provider>,
```

Reply:
updated